controllers/endpointconfig/endpointconfig_controller.go

```diff
@@ -67,10 +67,10 @@ func NewEndpointConfigReconciler(client client.Client, log logr.Logger, pollInte
 		Client:       client,
 		Log:          log,
 		PollInterval: pollInterval,
-		createSageMakerClient: func(awsConfig aws.Config) sagemakeriface.ClientAPI {
-			return sagemaker.New(awsConfig)
+		createSageMakerClient: func(cfg aws.Config) sagemakeriface.SageMakerAPI {
```

Comment:
This is repeated for the sake of mocking in unit tests. Is that what you mean?

Reply:
abstracting it would be nice 
but some of them use client wrappers 

https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/trainingjob/trainingjob_controller.go#L65

don't want to get into that as this seems unrelated code refactor